cw/cw305/capture.py

```diff
@@ -372,7 +418,12 @@ def capture_aes_fvsr_key_batch(ot, ktp, capture_cfg, scope_type):
             scope.num_segments = min(rem_num_traces, scope.num_segments_max)
             scope.arm()
 
-            # Start batch encryption.
+            # Start batch encryption. In order to increase capture rate, after the first batch
+            # encryption, the device will start automatically to generate random keys and plaintexts
+            # when this script is getting waves from the scope.
+            if is_first_batch:
+                ot.target.simpleserial_write("g", scope.num_segments_actual.to_bytes(4, "little"))
```

Comment:
I think you can just move this first call to `ot.target.simpleserial_write("g", scope.num_segments_actual.to_bytes(4, "little"))` to the outside of the loop. Then you can also remove the the `is_first_batch` variable completely.